pkg/curatedpackages/packagecontrollerclient_test.go

```diff
@@ -1,18 +1,33 @@
 package curatedpackages_test
 
 import (
+	"bytes"
 	"context"
+	_ "embed"
+	"encoding/base64"
 	"errors"
 	"fmt"
+	"os"
 	"testing"
 
 	"github.com/golang/mock/gomock"
 	. "github.com/onsi/gomega"
 
 	packagesv1 "github.com/aws/eks-anywhere-packages/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/config"
 	"github.com/aws/eks-anywhere/pkg/constants"
 	"github.com/aws/eks-anywhere/pkg/curatedpackages"
 	"github.com/aws/eks-anywhere/pkg/curatedpackages/mocks"
+	"github.com/aws/eks-anywhere/pkg/templater"
+)
+
+//go:embed config/awssecret.yaml
+var awsSecretYaml string
```

Comment:
I'm guessing we have this to be able to generate secrets in the tests. I would expect us to have files in testdata with the final expected result so we don't need to generate them on the fly. This makes the tests more robust: if there is a bug in the template and we use the same logic to generate the yaml in the code and in the tests, we won't catch the bug.